test/unit/component.js

```diff
@@ -82,142 +85,344 @@ describe('Component class', () => {
+            writable: false,
+            value: 'classes',
+          });
+          const expectedObj = {
+            viewData: 'viewData',
+            text: 'text',
+            model: 'model',
+            classes: 'classes',
+          };
+          assert.deepEqual(component.viewData, expectedObj);
+        });
+      });
+
+      describe('morphCallbacks', () => {
+        it('returns an object with the function onBeforeElUpdated', () => {
+          assert.instanceOf(component.morphCallbacks.onBeforeElUpdated, Function);
+        });
+
+        describe('onBeforeElUpdated()', () => {
+          it('returns false if fromEl\'s tagname is img and its source is toEl\'s data-src element', () => {
+            const fromEl = {tagName: 'IMG', src: 'data-src'};
+            const toEl = {
+              getAttribute(param) {
+                return param;
+              },
+            };
+            assert.equal(component.morphCallbacks.onBeforeElUpdated(fromEl, toEl), false);
+          });
+
+          it('returns true if fromEl\'s tagname is not img or its source is not toEl\'s data-src element', () => {
+            const fromEl = {tagName: 'not IMG', src: 'not data-src'};
+            const toEl = {
+              getAttribute(param) {
+                return param;
+              },
+            };
+            assert.equal(component.morphCallbacks.onBeforeElUpdated(fromEl, toEl), true);
+          });
+        });
       });
     });
 
-    describe('destroy()', () => {
-      it('delegates to view', () => {
-        const component = new Component({id: 1234});
-        component.view.destroy = sinon.spy();
-        component.destroy();
-        assert.calledOnce(component.view.destroy);
+    describe('"private" methods', () => {
+      describe('_userEvent()', () => {
+        it('logs to logger if debug is set to true', () => {
+          const infoSpy = sinon.spy();
+          const loggerStub = sinon.stub(logger, 'default').value({info: infoSpy});
+          component.globalConfig.debug = true;
+          component.typeKey = 'key';
+          component._userEvent('test');
+          assert.calledOnce(infoSpy);
+          assert.calledWith(infoSpy, 'EVENT: test (key)');
+          loggerStub.restore();
+        });
+
+        it('does not log if debug is set to false', () => {
+          const infoSpy = sinon.spy();
+          const loggerStub = sinon.stub(logger, 'default').value({info: infoSpy});
+          component.globalConfig.debug = false;
+          component._userEvent('test');
+          assert.notCalled(infoSpy);
+          loggerStub.restore();
+        });
+
+        it('calls event if the method passed is a function in the event', () => {
+          const eventSpy = sinon.spy();
+          component = Object.defineProperty(component, 'events', {
+            writable: false,
+            value: {test: eventSpy},
+          });
+
+          const isFunctionStub = sinon.stub(isFunction, 'default').returns(true);
```

Comment:
Is mocking `isFunction` is necessary? Is eventSpy not a function?

Reply:
isFunction is an import from the utils folder so I thought it would be safer to just not depend on it